test/nonStrict.test.ts

```diff
@@ -0,0 +1,155 @@
+import {
+  expectAcceptValuesImpure,
+  expectRejectValues,
+  objectAttributes,
+  st,
+} from './helpers'
+
+describe('nonStrict', () => {
```

Comment:
Similar to sloppyRecord tests, except that I've focused on aspects of non-strictness